crates/lib3h/src/engine/network_layer.rs

```diff
@@ -119,15 +120,23 @@ impl<'engine> GhostEngine<'engine> {
         // Note: use same order as the enum
         match request {
             transport::protocol::RequestToParent::ErrorOccured { uri, error } => {
-                self.network_connections.remove(uri);
-                error!("{} Network error from {} : {:?}", self.name, uri, error);
-                // Output a Lib3hToClient::Disconnected if it was the connection
-                if self.network_connections.is_empty() {
-                    let data = DisconnectedData {
-                        network_id: "FIXME".to_string(), // TODO #172
-                    };
+                if error.kind() == &transport::error::ErrorKind::Unbind {
+                    let data = UnboundData { uri: uri.clone() };
                     self.lib3h_endpoint
-                        .publish(Span::fixme(), Lib3hToClient::Disconnected(data))?;
+                        .publish(Span::fixme(), Lib3hToClient::Unbound(data))?;
+                } else {
+                    panic!("unhandled error {}", error);
```

Comment:
Is this intended behavior or a follow up is necessary?

Reply:
removed panic and added issue #391